libpod/image/pull.go

```diff
@@ -277,9 +302,11 @@ func cleanErrorMessage(err error) string {
 }
 
 // doPullImage is an internal helper interpreting pullGoal. Almost everyone should call one of the callers of doPullImage instead.
+// Note that the pullGoal is cleaned up.  Some of it's resources may not be available anymore.
```

Comment:
This works just fine, but it feels fragile. Anyone adding an extra error/return path to callers of `doPullImage` is fairly likely to overlook this. Requiring owners of a `pullGoal` to explicitly `defer goal.cleanUp()` would make this easier to see.

Reply:
I changed it as suggested. Very thoughtful!